iModelCore/ECPresentation/Source/Content/ContentDescriptorBuilder.cpp

```diff
@@ -345,8 +346,12 @@ struct ContentPropertiesAppender : ContentSpecificationsHandler::PropertyAppende
     ContentDescriptor::CalculatedPropertyField* CreateCalculatedPropertyField(ECClassCP ecClass, Utf8StringCR name, CalculatedPropertiesSpecificationCR spec,
         RelatedClassPathCR pathFromSelectToPropertyClass, RelationshipMeaning relationshipMeaning)
         {
+        PrimitiveType primitiveType;
+        if (ECObjectsStatus::Success != ValueHelpers::ParsePrimitiveType(primitiveType, spec.GetType().IsValid() ? spec.GetType().Value().c_str() : EC_PRIMITIVE_TYPENAME_STRING))
+            DIAGNOSTICS_HANDLE_FAILURE(DiagnosticsCategory::Content, "Provided type is not valid primitive type.");

```

Comment:
I think it would be better to do this:

```cpp
if (spec.GetType().IsValid())
    {
    if (SUCCESS != ValueHelpers::ParsePrimitiveType(primitiveType, spec.GetType().Value())
        DIAGNOSTICS_HANDLE_FAILURE(DiagnosticsCategory::Content, "Provided type is not valid primitive type for calculated fields.");
    }
else
    {
    primitiveType = PRIMITIVETYPE_String;
    }
```

This way we don't unnecessarily parse the default string type.

Reply:
Changed